(backport 87637/88593 to 4.0) modules: mbedtls: update to 3.6.3 and tf-m to 2.1.2

wearyzen opened the pull request:
Update Mbed TLS to 3.6.3 as it has CVE fixes.
Fixes #88435.
Related to #87637 .

wearyzen committed:
modules: mbedtls: update to 3.6.3

Update Mbed TLS to 3.6.3 as it has CVE fixes.

Signed-off-by: Sudan Landge <sudan.landge@arm.com>

dkalowsk commented:
Nothing against the PR, seems like a good addition.  Can you or @ceolin (or really someone from the security team) make sure the release notes are updated with the CVE mitigation notes?

tomi-font reviewed (changes requested):
needs changes in TF-M as well to work properly

ceolin commented:
@dkalowsk that is indeed a good thing to do. @wearyzen thanks for doing it. Are you willing to look TF-M changes as well ?

tomi-font commented:
I think that for 4.0 and 4.1 you can try to just point to my TF-M 2.1.2 update as it also makes sense to take it in, more than just the commits needed to work with Mbed TLS 3.6.3 (and the `main` branch in the TF-M fork should be compatible): https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/130

wearyzen committed:
modules: tf-m: update to 2.1.2

Update TF-M to 2.1.2 from version 2.1.1.
This is required to use MbedTLS 3.6.3.

Signed-off-by: Sudan Landge <sudan.landge@arm.com>

dkalowsk commented:
@mmahadevan108 once this PR (and it's dependencies) is merged, we should release 4.0.1.  Any objections?

tomi-font reviewed (changes requested):
same comments than on https://github.com/zephyrproject-rtos/zephyr/pull/88433#pullrequestreview-2783838322

tomi-font commented on doc/releases/release-notes-4.0.rst:
```diff
@@ -2,6 +2,48 @@
+Trusted Firmware-M (TF-M)
+*************************
+
+Updated TF-M to version 2.1.2 (from 2.1.1). The release notes can be found at:
```
```suggestion
TF-M was updated to version 2.1.2 (from 2.1.1). The release notes can be found at:
```
for a later force push

wearyzen replied:
Oops, missed it. Pushed it now just to be consistent with the other PR.

wearyzen commented:
@dkalowsk, @ceolin , I have updated the PR to include TF-M changes as well. Could you please have a look and help with the approval/merge if it looks good?
